inspire_disambiguation/api.py

```diff
@@ -76,34 +78,46 @@ def train_and_save_distance_model(
         ethnicity_model_path (str): Full path where ethnicity model is saved.
         save_distance_model_path (str): Full path where trained distance model
             will be saved.
-        sampled_pairs_size (int): Number of pairs to be generated for the training.
+        sampled_pairs_size (int): Number ożf pairs to be generated for the training.
             Note:
                 Must be multiple of 12.
     """
     LOGGER.info("Pulling training data from ES")
     start_time = datetime.now()
     curated_signatures = get_signatures(only_curated=True)
-    input_clusters = get_input_clusters(curated_signatures)
+    curated_signatures_dict = load_signatures(curated_signatures)
+    train_signatures_uuids = random.sample(list(curated_signatures_dict),
+                                           k=int(len(curated_signatures) * 0.8))
+    train_signatures = [curated_signatures_dict[uuid] for uuid in train_signatures_uuids]
+    test_signatures_uuids = set(list(curated_signatures_dict)) - set(list(curated_signatures_dict))
+    test_signatures = [curated_signatures_dict[uuid] for uuid in test_signatures_uuids]
+    input_clusters_train = get_input_clusters(train_signatures)
+    input_clusters_test = get_input_clusters(test_signatures)
     prepare_intput_time = datetime.now()
     LOGGER.info(
         "Preparing pairs from sampled data for training.",
         pairs_count=sampled_pairs_size,
     )
-    pairs = list(
-        sample_signature_pairs(curated_signatures, input_clusters, sampled_pairs_size)
+    pairs_train = list(
+        sample_signature_pairs(train_signatures, input_clusters_train, sampled_pairs_size)
     )
     prepare_pairs_time = datetime.now()
 
+    pairs_test = list(
+        sample_signature_pairs(test_signatures, input_clusters_test, sampled_pairs_size)
```

Comment:
* here you're testing only pairs `(test, test)`. Another interesting case would be `(train, test)`.
* getting that many pairs out of the testing set might be difficult. I think you should reduce the sampled pair size by the square of the ratio of sizes (so 25-fold if you do a 80/20% split).

Reply:
It's a good note but sampled_pairs_size must be also a multiple of 12